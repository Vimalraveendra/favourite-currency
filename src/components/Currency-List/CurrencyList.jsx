import React from "react";

class CurrencyList extends React.Component {
  constructor() {
    super();
    this.state = {
      base: "EUR",
      amount: 1,
      convertedTo: "USD",
      currencies: [],
      rates: [],
      result: ""
    };
  }
  componentDidMount() {
    fetch(`http://api.nbp.pl/api/exchangerates/tables/a/today/?format=JSON`)
      .then(response => response.json())
      .then(data => {
        data = data[0].rates;
        let currencyresult = [];
        // let baseValue = "";
        // let convertedValue = "";
        data.forEach(item => {
          currencyresult.push(item.code);
        });
        //   if (item.code === base) {
        //     baseValue = item.mid;
        //     console.log("base", baseValue);
        //   }
        //   if (item.code === convertedTo) {
        //     convertedValue = item.mid;
        //     console.log("value", convertedValue);
        //   }
        //   let result = amount * convertedValue;
        //   console.log("hello", result);
        // });
        this.setState({ currencies: currencyresult, rates: data });
      });
  }
  render() {
    console.log("currencies", this.state.currencies);
    console.log("rates", this.state.rates);
    return (
      <div className="homepage">
        <h1>Favorite Currency</h1>
      </div>
    );
  }
}

export default CurrencyList;
