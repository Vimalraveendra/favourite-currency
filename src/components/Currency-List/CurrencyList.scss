.title {
  font-size: 2rem;
  font-weight: 500;
  font-family: "Monoton", cursive;
  color: red;
}

@media screen and (max-width: 600px) {
  .title {
    font-size: 1.5rem;
  }
}
