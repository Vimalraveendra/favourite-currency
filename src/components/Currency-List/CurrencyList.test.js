import React from 'react';
import{shallow} from 'enzyme';
import CurrencyList from './CurrencyList';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
const mockStore = configureStore();
const store = mockStore();



it("should render Currency component  with default props and redux store", () => {
    let wrapper = shallow(
        <Provider store={store}>
        <CurrencyList/>
        </Provider>
        );
    expect(wrapper).toMatchSnapshot();
  });