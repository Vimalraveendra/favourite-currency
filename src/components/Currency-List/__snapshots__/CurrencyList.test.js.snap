// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Currency component  with default props and redux store 1`] = `ShallowWrapper {}`;
